03_launch_mgmt_cluster.sh

```diff
@@ -219,6 +224,39 @@ EOF
   popd
 }
 
+#
+# Launch and configure fakeIPA
+#
+function launch_fakeIPA() {
+  # Create a folder to host fakeIPA config and certs
+  mkdir -p /opt/metal3-dev-env/fake-ipa
+  if [[ "${EPHEMERAL_CLUSTER}" == "kind" ]]; then
+    if [[ "${IRONIC_TLS_SETUP}" == "true" ]]; then
+      cp "${IRONIC_CACERT_FILE}" "/opt/metal3-dev-env/fake-ipa/ironic-ca.crt"
+    fi
+  else
+    # wait for ironic to be running to ensure ironic-cert is created 
+    kubectl -n baremetal-operator-system wait --for=condition=available deployment/baremetal-operator-ironic --timeout=900s
+    # Extract ironic-cert to be used inside fakeIPA for TLS 
+    kubectl -n legacy get secret -n baremetal-operator-system ironic-cert -o json -o=jsonpath="{.data.ca\.crt}" | base64 -d > /opt/metal3-dev-env/fake-ipa/ironic-ca.crt
+  fi
+  # Create fake IPA custom config
+  cat << EOF > "${WORKING_DIR}/fake-ipa/config.py"
+FAKE_IPA_API_URL = "https://${CLUSTER_BARE_METAL_PROVISIONER_IP}:${IRONIC_API_PORT}"
+FAKE_IPA_INSPECTION_CALLBACK_URL = "${IRONIC_URL}/continue_inspection"
+FAKE_IPA_ADVERTISE_ADDRESS_IP = "${EXTERNAL_SUBNET_V4_HOST}"
+FAKE_IPA_INSECURE = ${FAKE_IPA_INSECURE:-False}
+FAKE_IPA_CAFILE = "${FAKE_IPA_CAFILE:-/root/cert/ironic-ca.crt}"
+FAKE_IPA_MIN_BOOT_TIME = ${FAKE_IPA_MIN_BOOT_TIME:-20}
+FAKE_IPA_MAX_BOOT_TIME = ${FAKE_IPA_MAX_BOOT_TIME:-30}
+EOF
+  # shellcheck disable=SC2086
+  sudo "${CONTAINER_RUNTIME}" run -d --net host --name fake-ipa ${POD_NAME_INFRA} \
+    -v "/opt/metal3-dev-env/fake-ipa":/root/cert -v "/root/.ssh":/root/ssh \
+    -e CONFIG='/root/cert/config.py' \
+    "${FAKE_IPA_IMAGE}"
+}
```

Comment:
I am thinking if we should add this function in a separate script file for better management and keeping 03_ less consumed?

Reply:
which function you mean ? this container run ?

Reply:
`launch_fakeIPA()`

Reply:
I am just following the old structure all the launch function are defined here ..

Reply:
You are correct, indeed we have quite a few of those launch functions. However, most of the launch functions are indeed a part of the 03 normal workflow (they are almost always called) whereas I feel this fake_ipa function is not. That being said, this is not a functional requirement. So I am ok if you insist to keep it here.